jetty-webapp/src/main/java/org/eclipse/jetty/webapp/WebInfConfiguration.java

```diff
@@ -507,8 +514,18 @@ public void makeTempDirectory(File parent, WebAppContext context)
         }
         else
         {
-            //ensure file will always be unique by appending random digits
-            tmpDir = Files.createTempDirectory(parent.toPath(), temp).toFile();
+            Path parentPath = parent.toPath();
+            FileStore fileStore = Files.getFileStore(parentPath);
+            if (fileStore.supportsFileAttributeView(PosixFileAttributeView.class))
+            {
+                String workDirPerms = context.getTempDirectoryPosixPermissions();
+                Set<PosixFilePermission> permSet = PosixFilePermissions.fromString(workDirPerms);
+                tmpDir = Files.createTempDirectory(parentPath, temp, PosixFilePermissions.asFileAttribute(permSet)).toFile();
+            }
```

Comment:
I'm not sure we need to create a lot of new API for this.  Can we just do it with attributes: context, server then system

```suggestion
            if (fileStore.supportsFileAttributeView(PosixFileAttributeView.class))
            {
                String workDirPerms = (String) context.getAttribute(TEMP_DIRECTORY_POSIX_PERMISSIONS);
                if (workDirPerms == null) 
                    workDirPerms = (String) context.getServer().getAttribute(TEMP_DIRECTORY_POSIX_PERMISSIONS);
                if (workDirPerms == null) 
                    workDirPerms = (String) System.getProperty(TEMP_DIRECTORY_POSIX_PERMISSIONS, "rwx------");
                Set<PosixFilePermission> permSet = PosixFilePermissions.fromString(workDirPerms);
                tmpDir = Files.createTempDirectory(parentPath, temp, PosixFilePermissions.asFileAttribute(permSet)).toFile();
            }
```

We've changed tmp dir stuff a number of times... once a thing is said in API it is forever!   So I think we just lean towards to loose attributes rather than API.

Reply:
What's the String you see for `TEMP_DIRECTORY_POSIX_PERMISSIONS` ?

Reply:
The biggest issue with attributes is making them friendly for start modules.

Reply:
Also, where does `TEMP_DIRECTORY_POSIX_PERMISSIONS` exist?
Not the `WebAppContext`, how about `IO` ? or the `Server`?

Reply:
I know this is closed, but will answer so record my thoughts.

I see these attributes as hints to the implementation, so the string should be "org.eclipse.jetty.webapp.WebInfConfiguration.TEMP_DIRECTORY_POSIX_PERMISSIONS" and defined as a constant in WebInfConfiguration.

It is easy to make friendly for start modules, because we write the XML for those modules, thus we write the XML like:
```xml
  <Call name="setAttribute">
    <Arg>org.eclipse.jetty.webapp.WebInfConfiguration.TEMP_DIRECTORY_POSIX_PERMISSIONS</arg>
    <Arg><Property name="jetty.webapp.tmpDirPosixPermissions"/></Arg>
  </Call>
```

and then we just add to the ini-template of `webapp.mod`:
```
## Set the permissions of temporary directory on POSIX file systems
# jetty.webapp.tmpDirPosixPermissions=rwx------
```

Ultimately, this is no different at the start module level than if we had dedicated APIs